open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/data-platform-connectors/cassandra-open-metadata-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/cassandra/CassandraStoreConnector.java

```diff
@@ -0,0 +1,189 @@
+
+        if (omrsAuditLog != null) {
+            auditLog = CassandraConnectorAuditCode.CONNECTOR_INITIALIZING;
+            omrsAuditLog.logRecord(
+                    actionDescription,
+                    auditLog.getLogMessageId(),
+                    auditLog.getSeverity(),
+                    auditLog.getFormattedLogMessage(),
+                    null,
+                    auditLog.getSystemAction(),
+                    auditLog.getUserAction());
+        }
+
+        if (endpoint != null) {
+            serverAddress = endpoint.getAddress();
+
+            if (serverAddress != null) {
+                log.info("The connecting cassandra cluster server address is: {}.", serverAddress);
+
+            } else {
+                log.error("Errors in the Cassandra server configuration. The address of the server cannot be extracted.");
+                if (omrsAuditLog != null) {
+                    auditLog = CassandraConnectorAuditCode.CONNECTOR_SERVER_CONFIGURATION_ERROR;
+                    omrsAuditLog.logRecord(actionDescription,
+                            auditLog.getLogMessageId(),
+                            auditLog.getSeverity(),
+                            auditLog.getFormattedLogMessage(),
+                            null,
+                            auditLog.getSystemAction(),
+                            auditLog.getUserAction());
+                }
+            }
+        } else {
+            log.error("Errors in Cassandra server address. The endpoint containing the server address is invalid.");
+            if (omrsAuditLog != null) {
+                auditLog = CassandraConnectorAuditCode.CONNECTOR_SERVER_ADDRESS_ERROR;
+                omrsAuditLog.logRecord(actionDescription,
+                        auditLog.getLogMessageId(),
+                        auditLog.getSeverity(),
+                        auditLog.getFormattedLogMessage(),
+                        null,
+                        auditLog.getSystemAction(),
+                        auditLog.getUserAction());
+            }
+        }
+
+        startCassandraConnection();
+
+        if (cluster.getClusterName().equals(clusterName) && omrsAuditLog != null)
+        {
+            auditLog = CassandraConnectorAuditCode.CONNECTOR_INITIALIZED;
+            omrsAuditLog.logRecord(actionDescription,
+                    auditLog.getLogMessageId(),
+                    auditLog.getSeverity(),
+                    auditLog.getFormattedLogMessage(),
+                    null,
+                    auditLog.getSystemAction(),
+                    auditLog.getUserAction());
+        }
+    }
+
+    /**
+     * Set up the Cassandra Cluster Connection
+     */
+    public void startCassandraConnection() {
+
+        this.cluster = Cluster.builder()
+                .addContactPoint(serverAddress)
+                .withClusterName(clusterName)
+                .withCredentials(username, password)
+                .build();
+        session = cluster.connect();
```

Comment:
I think the session should be closed/cleaned up before shiudown

Reply:
True. 👍